lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp

```diff
@@ -660,6 +660,60 @@ int amf_app_handle_pdu_session_response(
 
   return rc;
 }
+/****************************************************************************
+ **                                                                        **
+ ** Name:    convert_ambr()                                                **
+ **                                                                        **
+ ** Description: Converts the session ambr format from                     **
+ **  one defined in create_pdu_session_response to one defined in          **
+ **  pdu_session_estab_accept message                                      **
+ **                                                                        **
+ ** Inputs:  pdu_ambr_response_unit, pdu_ambr_response_value               **
+ **          ambr_unit, ambr_value                                         **
+ **                                                                        **
+ **      Return:   void                                                    **
+ **                                                                        **
+ ***************************************************************************/
+void convert_ambr(
+    const uint32_t* pdu_ambr_response_unit,
+    const uint32_t* pdu_ambr_response_value, uint8_t* ambr_unit,
+    uint16_t* ambr_value) {
+  int count = 1;
+  if (*pdu_ambr_response_unit == KBPS) {
+    count = 2;
+  }
+  uint32_t temp_pdu_ambr_response_value = *pdu_ambr_response_value;
+  if (temp_pdu_ambr_response_value / 1000 != 0) {
```

Comment:
why do you need this if statement? should not this block simply be:
``` 
while (temp_pdu_ambr_response_value >= AMBR_UNIT_CONVERT_THRESHOLD) {
     temp_pdu_ambr_response_value = (temp_pdu_ambr_response_value / 1000);
     count++;
}
```

Reply:
Initially I implemented a do while loop, hence I wrote that if statement there. I'll remove it.

Reply:
@ulaskozat  `if (temp_pdu_ambr_response_value / 1000 != 0)` is a necessary check. The main reason we raised this PR is to fix the bug in zenhub task : #10220. In this bug is there because we are not converting the sessiond itti response ambr format. We are sending absolute value and ambr_unit as 0. In pcap we can see that this is treated as illegal value. This means that anything less than kbps is illegal. But if I remove the if condition it will start making all kbps values upto 65535 as illegal value.

Reply:
@ulaskozat if (temp_pdu_ambr_response_value / 1000 != 0) is a necessary check. The main reason we raised this PR is to fix the bug in zenhub task : #10220. This bug is there because we are not converting the sessiond itti response ambr format. We are sending absolute value and ambr_unit as 0. In pcap we can see that this is treated as illegal value. This means that anything less than kbps is illegal. But if I remove the if condition it will start making all kbps values upto 65535 as illegal value.

Reply:
So `case 1` on line 694 should never occur? This is absolutely not clearly documented anywhere in this function implementation. If KBPS is the minimum unit, than the logic should check `if (*pdu_ambr_response_unit == BPS)` on line 682 instead and if so divide it by 1000. The count then should be initialized as 1 and original case 1 can be KBPS, case 2 can be MBPS, and so on. Also default case looks like an error (this should not happen at all given input is uint32_t) and should be handled accordingly.

Reply:
if `pdu_ambr_response_value` is less that 1kbps and with units bps, you need to round it up to 1Kbps. so you need to cover that as a special case.

Reply:
@ulaskozat yes, default case will not occur. But other than that the logic works fine. Because if we send anything less than 1000 BPS eg: `987 BPS`, it will be converted as `987` and uint will be `Value not used`. Hence it will be forwarded to gnb as illegal value [Suggesting that operator/user should check subscription data], which is the case. As ambr value was subscribed like that (In subscriberDB). But if we increase the value by ourself we are over ruling the subscription data as well as policy information. As this conversion takes place after validation by sessiond (Validation of subscription data vs policy).

Reply:
@Lkishor123 : Lets fix it for 1Kbps.

Reply:
@Lkishor123 as long as the documentation clearly states that AMBR is set minimum at 1Kbps regardless of policy, it is not an issue. Note that we currently use TC and it cannot really enforce anything smaller than 10Kbps.

Reply:
@ulaskozat understood, I'll update it.

Reply:
@ulaskozat, I've updated the code, please review.

Reply:
@ulaskozat, I've updated the code and rebased with latest master, please review.